src/template.rs

```diff
@@ -220,11 +262,34 @@ impl<'a> Renderer<'a> {
                 pages: match &info.index_site_pages {
                     Some(templates) => {
                         if templates.contains(&tpl) {
-                            crate::content::all_pages(
-                                self.content_dir.clone(),
-                                self.show_unpublished,
-                                self.disable_cache,
-                            )?
+                            let mut glob_pattern: Option<Vec<String>> = None;
+                            let template_meta = self.template_meta.get(&tpl);
+                            if let Some(val) = template_meta {
+                                if val.read_pages_glob.is_some() {
+                                    glob_pattern =
+                                        Some(val.read_pages_glob.as_ref().unwrap().to_owned());
+                                }
+                            }
+                            match glob_pattern {
+                                Some(pattern) => {
+                                    let pages = crate::content::get_pages_by_glob(
+                                        self.content_dir.clone(),
+                                        pattern,
+                                        self.show_unpublished,
+                                    );
+                                    match pages {
+                                        Ok(val) => val,
+                                        Err(err) => {
+                                            bail!("Error parsing glob in template \"{tpl}\": {err}")
```

Comment:
If I've counted this right, it's _eleven_ levels of indent, and the function has grown to 90 lines long.  I realise you want to get this out quickly so maybe we address that in a follow-up pass, but at some point it would be good to break out a helper function or two.